send-transaction-service/src/transaction_client.rs

```diff
@@ -275,18 +266,18 @@ where
                 tpu_peers,
             };
         let config = Self::create_config(identity, leader_forward_count as usize);
-        let handle = runtime_handle.spawn(ConnectionWorkersScheduler::run(
-            config,
-            Box::new(leader_updater.clone()),
-            receiver,
+        let scheduler = ConnectionWorkersScheduler::new(Box::new(leader_updater), receiver);
+        // leaking handle to this task, as it will run until the end of life of the process
```

Comment:
It might be cancelled as well when, for example, validator identity is updated

Reply:
Fixed in so the metric task gets restarted as well. Do not think patching it into tpu_client_next body would be much better. https://github.com/anza-xyz/agave/pull/5338/commits/fc19d78ae06dcd99cc6d8b672c2744db50221479